### PR TITLE
[opctwin] - fixes to  run as windows edge-module

### PR DESCRIPTION
**Pull request by cristipogacean**
* fixed the connection string for endpoints twin management using the Iot EdgeHub 
* fixed the unix socket based (unix://..) edgelet communication on windows

### CODE REVIEW

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs [hunk 1]
```diff
@@ -339,8 +339,13 @@ private class TwinHost : IDisposable, IProcessControl, IModuleConfig {
                         throw new InvalidConfigurationException(
                             "Missing IOTEDGE_IOTHUBHOSTNAME variable in environment");
                     }
-                    cs = $"HostName={hostName};DeviceId={endpointId};SharedAccessKey={secret}";
                     var edgeName = Environment.GetEnvironmentVariable("IOTEDGE_GATEWAYHOSTNAME");
+                    if (string.IsNullOrEmpty(edgeName)) {
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs]: isn't the if reversed here? edgeName != null etcetc

File: common/src/Microsoft.Azure.IIoT.Core/src/Extensions/UriEx.cs [hunk 2]
```diff
@@ -90,6 +90,10 @@ public static class UriEx {
                 fileDevice = localPath.Substring(0, index + 5);
                 pathAndQuery = localPath.Substring(index + 5);
             }
+            else if ((index = localPath.IndexOf("sock", StringComparison.InvariantCultureIgnoreCase)) != -1) {
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.Core/src/Extensions/UriEx.cs]: better here to define strings (e.g. ".sock")as consts and derive the index modifier from constants

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs [hunk 3]
```diff
@@ -353,7 +358,14 @@ private class TwinHost : IDisposable, IProcessControl, IModuleConfig {
                         throw new InvalidConfigurationException(
                             "Missing HostName in connection string");
                     }
-                    cs = $"HostName={hostName};DeviceId={endpointId};SharedAccessKey={secret}";
+
+                    if (!lookup.TryGetValue("GatewayHostName", out var edgeName) ||
```
Comment (mregen) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs]: same as above, reversed logic?

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs]: indeed, that is fixed now

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs (hunk 3 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge/src/Supervisor/SupervisorServices.cs]: indeed, this is fixed now

File: common/src/Microsoft.Azure.IIoT.Core/src/Extensions/UriEx.cs (hunk 2 above)
Comment (cristipogacean) [reply common/src/Microsoft.Azure.IIoT.Core/src/Extensions/UriEx.cs]: the "socket syntax is in linux is ".sock" and in windows is "/sock", therefore the "sock" check is enough for both, not that strict nevertheless...

**Pull request merged**